tests/v2/integration/steveapi/steve_api_test.go

```diff
@@ -451,6 +453,246 @@ func (s *SteveAPITestSuite) TestList() {
 				{"name": "test5", "namespace": "test-ns-1"},
 			},
 		},
+		{
+			description: "user:user-a,namespace:none,query:filter=metadata.name=test1",
+			user:        "user-a",
+			namespace:   "",
+			query:       "filter=metadata.name=test1",
+			expect: []map[string]string{
+				{"name": "test1", "namespace": "test-ns-1"},
+				{"name": "test1", "namespace": "test-ns-2"},
+				{"name": "test1", "namespace": "test-ns-3"},
+				{"name": "test1", "namespace": "test-ns-4"},
+				{"name": "test1", "namespace": "test-ns-5"},
+			},
+		},
+		{
+			description: "user:user-a,namespace:none,query:filter=metadata.name=test6",
+			user:        "user-a",
+			namespace:   "",
+			query:       "filter=metadata.name=test6",
+			expect:      []map[string]string{},
+		},
+		{
+			description: "user:user-a,namespace:test-ns-1,query:filter=metadata.name=test1",
+			user:        "user-a",
+			namespace:   "test-ns-1",
+			query:       "filter=metadata.name=test1",
+			expect: []map[string]string{
+				{"name": "test1", "namespace": "test-ns-1"},
+			},
+		},
+		{
+			description: "user:user-a,namespace:none,query:sort=metadata.name",
+			user:        "user-a",
+			namespace:   "",
+			query:       "sort=metadata.name",
+			expect: []map[string]string{
+				{"name": "test1", "namespace": "test-ns-3"},
+				{"name": "test1", "namespace": "test-ns-5"},
+				{"name": "test1", "namespace": "test-ns-2"},
+				{"name": "test1", "namespace": "test-ns-4"},
+				{"name": "test1", "namespace": "test-ns-1"},
+				{"name": "test2", "namespace": "test-ns-1"},
+				{"name": "test2", "namespace": "test-ns-5"},
+				{"name": "test2", "namespace": "test-ns-4"},
+				{"name": "test2", "namespace": "test-ns-2"},
+				{"name": "test2", "namespace": "test-ns-3"},
+				{"name": "test3", "namespace": "test-ns-3"},
+				{"name": "test3", "namespace": "test-ns-2"},
+				{"name": "test3", "namespace": "test-ns-5"},
+				{"name": "test3", "namespace": "test-ns-1"},
+				{"name": "test3", "namespace": "test-ns-4"},
+				{"name": "test4", "namespace": "test-ns-1"},
+				{"name": "test4", "namespace": "test-ns-4"},
+				{"name": "test4", "namespace": "test-ns-3"},
+				{"name": "test4", "namespace": "test-ns-2"},
+				{"name": "test4", "namespace": "test-ns-5"},
+				{"name": "test5", "namespace": "test-ns-1"},
+				{"name": "test5", "namespace": "test-ns-3"},
+				{"name": "test5", "namespace": "test-ns-4"},
+				{"name": "test5", "namespace": "test-ns-2"},
+				{"name": "test5", "namespace": "test-ns-5"},
+			},
+		},
```

Comment:
Will this always be the correct order since we are only sorting on metadata.name

Reply:
Theoretically there is no guarantee, but practically there is because the resource order from etcd is deterministic and so the result of the sort is also deterministic. But I agree this is a little wonky, I think I'll change this to ignore the irrelevant  field for these types of cases.